cpp/include/wholememory/wholememory.h

```diff
@@ -148,6 +153,13 @@ wholememory_error_code_t wholememory_communicator_get_rank(int* rank, wholememor
  */
 wholememory_error_code_t wholememory_communicator_get_size(int* size, wholememory_comm_t comm);
 
+bool wholememory_communicator_is_bind_to_nvshmem(wholememory_comm_t comm);
+
+wholememory_error_code_t wholememory_communicator_set_preferred_distributed_backend(
+  wholememory_comm_t comm, wholememory_distributed_backend_t distributed_backend);
+
+wholememory_distributed_backend_t wholememory_communicator_get_preferred_distributed_backend(
+  wholememory_comm_t comm);
```

Comment:
What do you mean by `preferred` here? Does this mean WholeMemory will try to set back-end based on the input argument but may end up setting the back-end to something else? If this is indeed the case, some documentation will help. If this functions returns on error code if the back-end can't be set as specified, then, better remove `preferred` from the function name to avoid confusion.

Reply:
change `preferred_distributed_backend` to `distributed_backend`